plugins/autoid/plugin.js

```diff
@@ -158,6 +160,208 @@
         return !(/\w/.test(character));
       }
 
+      // This modifies the dialog for the 'link' plugin to allow the option of
+      // internal document linking to autoid headings. Original 'link' plugin
+      // code can be found at:
+      // https://github.com/ckeditor/ckeditor-dev/tree/master/plugins/link
+
+      function modifyLinkDialog(ev) {
+        var dialog = ev.data;
+
+        if (dialog.name == 'link') {
+          var def = ev.data.definition,
+            infoTab = def.contents[0],
+            linkTypeTab = infoTab.elements[1];
+
+          // Add option for heading links to linkType dropdown
+          linkTypeTab.items.push(['Link to heading in the text', 'heading']);
+
+          // Add additional content to 'Link Info' tab for heading links
+          infoTab.elements.push(createHeadingLinkContent());
+
+          // Modify linkType's 'onChange' function to accomodate new heading option
+          linkTypeTab.onChange = modifiedLinkTypeChanged;
+
+          // Modify main dialog 'onOk' function to handle heading data
+          def.onOk = modifiedOnOk;
+        }
+      }
+
+      function modifiedLinkTypeChanged() {
```

Comment:
Yes.  Actually this function would not be needed at all if the heading link options were in an entirely separate tab.  I do, however, think that that would make for a less intuitive user interface, since all the other link types are accessed through a single tab.